### PR TITLE
Creating dummy data in python using faker package

### LINKED ISSUE #4951: [Language]Creating Dummy Data In Python Using Faker Package
### Topic Suggestion
 Creating Dummy Data In Python Using Faker Package

#### Proposed article introduction
It's important to have fake data for a variety of reasons. It can be difficult to find the data you need in a specific format. The Faker package in Python is used to create dummy data, which is the subject of this article.

Dummy data is a useful tool for testing and evaluating software and hardware. You can use it to see how your code responds to various inputs after you've finished developing it. Faker, a Python package, can be used to create dummy data. An open-source library that generates dummy data of various types is available here.
#### Key takeaways 
- Why do we need dummy data
- Installing the Faker Package
- How to create and initialize a Faker Generato
- Providers
- Command-line usage of Faker Package

#### Article quality
The reader will discover how to use Python's Faker library to generate various forms of data. The reader investigates how to generate names, biographical profiles, and currency-related data. The reader will also learn how to replicate the same dummy data and how to generate unique data. We will also look at the providers and discover that it is possible to create data that is specific to a specific location.

#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Dianapeter**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4951

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+### Introduction
+For various reasons, it's essential to have phoney data on hand. In this post, dummy data is generated using the Python Faker library. It's critical to test and evaluate software and hardware with dummy data. It's possible to test your code by running it through various scenarios. Faked data can be easily generated with the Python function Faker.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:2]: `For various reasons, it's essential to have phoney data on hand`

??

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+For testing and operational reasons, dummy data is used. Your code can be tested against a variety of different types of stimuli.
+### Prerequisites
+1. Before proceeding with this tutorial, you should know the python programming language.
+2. Have a python IDE or follow this [documentation](https://www.python.org/) to install.
+
+> Setup Faker with the pip command:
+
+```Python
+Pip install Faker
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:12]: Pip?

It should be `pip`

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+```
+
+### Listed below are methods and types of dummy data
+#### 1. Faker generators can be created and initialized
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:16]: This is not a method right? It's common to all of the them. So, don't add it as a subheading

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ourFake.text()
+```
+Output:
+```bash
+About eight four they wife.
+Force finally very summer. Occur, you shake through physical reveal. Require player yeah magazine thank bank traditional.
+```
+For addresses we use `address()` and for names we use `name()` methods.
+#### 3. Creating same dummy data
+You may wish to use the same collection of data again. Seeding the generator is a viable option. The following code generates dummy data using the seed() method:
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:34]: Explain what seeding is, and where it is used.

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+```python
+# generates 10 unique texts 
+texts = [ourFake.unique.name() for b in range(10)]
+```
+#### 5. Command-line usage of Faker package
+Command-line invocation of the faker package allows you to directly type in the command prompt code.
+#### 6. Providers
+For the most part, these features are bundled into service providers.
+1. Standard Providers- You can find more information [here](https://faker.readthedocs.io/en/stable/providers.html).
+2. Community Providers- [This](https://faker.readthedocs.io/en/stable/communityproviders.html) site has a wealth of data on locally developed service providers.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:55]: Explain them here. Don't just add reference link

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+प्रबोध खान
+रायन कृष्णन
+लाल, नरेन्द्र
+काले, फ़रहान
+प्रभाकर भंडारी
+कुलकर्णी, तृष्णा
+सरस्वती बालासुब्रमणियम
+गौतम दुआ
+विजय कृष्णा
+```
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:77]: I'm not sure if Section website supports other languages' alphabets. I would suggest you to replace with some other language that contains English alphabets.

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+### Other dummy data creation methods
+They are as follows:
+
+1. In Python, Numpy's Random module can be accessed: Pseudorandom numbers can be generated with the random package. Many functions are available, including `rand()`, `randint()`, and many more.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:81]: Add examples

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+2. Fauxfactory: To test your code quickly, you can use this anytime. [Here](https://fauxfactory.readthedocs.io/en/latest/#), you may learn more about it.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:82]: Explain more about it

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+###  A few more words regarding dummy data
+As a result of dummy variables, a single regression equation might represent multiple groups. The necessity for different equations for each subgroup is eliminated. Put another way, and dummy variables are like switches in an equation that can enable or disable other variables.
+#### Effect of a dummy variable
+When the coefficient of an explanatory or independent variable is set to zero, it cannot affect the dependent variable. Still, the intercept will be affected when its coefficient is set to 1.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:86]: What is regression? 

`For various reasons, it's essential to have phoney data on hand` -> What are they?

Explain in-detail

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+Dummy variables are created via one-hot encoding. One of these variables must be used as the starting point to avoid perfect multicollinearity.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:88]: What is one hot encoding? What is multicollinearity?

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+With this encoding method, the categorical variable is converted into a set of binary variables. One-hot encoding employs N binary variables for N categories in a variable. Dummy encoding is an improvement over one-hot encoding, but it's not much of an improvement.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:90]: Add example

File: content/articles/creating-dummy-data-in-python-using-faker-package/index.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+#### Dummy variable trap
+Dummy variable traps can be avoided if you have many characteristics that are highly connected (Multicollinear). When categorical data is encoded using one-hot encoding, we can predict one dummy variable (attribute) using other dummy variables.
```
Comment (srishilesh) [inline content/articles/creating-dummy-data-in-python-using-faker-package/index.md:92]: Explain more about this. Tell where we face this scenario. Add examples.

**Review by srishilesh — CHANGES_REQUESTED** [review]
@Dianapeter 

This PR looks too low on the technical content.

There are several points that needs to be improved on, I've suggested a few of them. These were not explained or they lacked clarity. You can add examples to explain them.

Adding a example scenario of where we use them, would definitely help.

See how you can add value from what is not there in the Faker python documentation.

Since, this would require several revisions, we would be closing this PR for now. You can work on your forked repo.

You can create a new PR with all the changes, and ping me there.

Thanks.

**Pull request closed without merging**